contracts/Pod.sol

```diff
@@ -68,79 +81,121 @@ contract Pod is Initializable, ReentrancyGuardUpgradeSafe, ERC777UpgradeSafe, Ba
         return unlockTimestamp[_user];
     }
 
-    function deposit(uint256 _amount) external nonReentrant returns (uint256) {
+    function mintShares(address _receiver, uint256 _amount) external nonReentrant returns (uint256) {
+        // Buy Tickets for caller
         address _sender = _msgSender();
-        Ticket _ticket = ticket();
-        IERC20 _assetToken = yieldService().token();
+        _buyTickets(_sender, _amount);
 
-        // Collect collateral from caller
-        _assetToken.transferFrom(_sender, address(this), _amount);
-
-        // Calculate 
+        // Calculate & Mint Pod-Shares
         uint256 _shares = FixedPoint.divideUintByMantissa(_amount, exchangeRateMantissa());
-
-        // Transfer collateral to Prize Pool for Tickets
-        _assetToken.approve(address(_ticket), _amount);
-        _ticket.mintTickets(_amount);
-
-        // Mint Pod-Shares
-        _mint(_sender, _shares);
+        _mint(_receiver, _shares);
 
         // Log event
-        emit PodDeposit(_sender, _amount, _shares);
-
-        // Return the amount of Shares transferred
+        emit PodDeposit(_sender, _receiver, _amount, _shares);
         return _shares;
     }
 
-    function redeemSharesInstantly(uint256 _shares) external nonReentrant returns (uint256) {
+    function redeemSharesInstantly(address _from, uint256 _shares) external nonReentrant returns (uint256) {
```

Comment:
Let's be consistent with ERC777 here, regarding the function signatures. 

Instead of the above, we'd have:

`redeemSharesInstantly(uint256)`

and

`operatorRedeemSharesInstantly(address,uint256)`

You'll note that's the convention used in ERC777.